unittest/buildtool/versions_command_test.py

```diff
@@ -0,0 +1,290 @@
+import buildtool.versions_commands
+
+from buildtool.versions_commands import (
+    VersionsBuilder,
+    PublishVersionsCommand,
+    MAXIMUM_SPINNAKER_RELEASE_COUNT,
+)
+
+
+class TestVersionsBuilder(BaseTestFixture):
+    """Test Version Builder."""
+
+    def setUp(self):
+        super().setUp()
+        self.test_root = os.path.join(self.base_temp_dir, self._testMethodName)
+
+    def test_build_single_version_dict(self):
+        """A single release version"""
+        builder = VersionsBuilder()
+
+        with patch("buildtool.versions_commands.now") as mock_now:
+            mock_now.return_value = 1234567890000
+            got = builder.build_single_release_dict(
+                "1.29.0",
+                "1.50.0",
+            )
+
+        expected = {
+            "alias": "v1.29.0",
+            "changelog": f"{SPINNAKER_CHANGELOG_BASE_URL}/1.29.0-changelog/",
+            "lastUpdate": 1234567890000,
+            "minimumHalyardVersion": "1.50.0",
+            "version": "1.29.0",
+        }
+        self.assertDictEqual(expected, got)
+
+    def test_get_major_minor_patch_version(self):
+        """A version string should get split into an array of strings: major, minor, patch."""
+        version = "1.2.3"
+        expected = ["1", "2", "3"]
+        builder = VersionsBuilder()
+        got = builder.get_major_minor_patch_version(version)
+        self.assertListEqual(expected, got)
+
+    def test_add_or_replace_release_new_minor(self):
+        """A new minor version should be added to versions.yml and the list of
+        versions should increase by one."""
+        builder = VersionsBuilder()
+
+        release_100 = builder.build_single_release_dict("1.0.0", "1.50.0")
+        release_110 = builder.build_single_release_dict("1.1.0", "1.50.0")
+        release_120 = builder.build_single_release_dict("1.2.0", "1.50.0")
+        releases = [release_100, release_110, release_120]
+
+        release_140 = builder.build_single_release_dict("1.4.0", "1.50.0")
+
+        got = builder.add_or_replace_release(release_140, releases)
+
+        self.assertEqual(len(got), 4)  # count of releases has increased
+        self.assertDictEqual(got[-1], release_140)
+
+    def test_add_or_replace_release_new_patch(self):
+        """A new patch version should replace existing in versions.yml and
+        there should remain our accepted maximum versions in the list."""
+        builder = VersionsBuilder()
+
+        release_100 = builder.build_single_release_dict("1.0.0", "1.50.0")
+        release_110 = builder.build_single_release_dict("1.1.0", "1.50.0")
+        release_120 = builder.build_single_release_dict("1.2.0", "1.50.0")
+        releases = [release_100, release_110, release_120]
+
+        release_111 = builder.build_single_release_dict("1.1.1", "1.50.0")
+
+        got = builder.add_or_replace_release(release_111, releases)
+
+        self.assertEqual(len(got), 3)  # count of releases has not increased
```

Comment:
This'll break if the max changes...maybe that's OK.  Maybe add an assertion above about `len(releases) == MAXIMUM_SPINNAKER_RELEASE_COUNT` and use `len(releases)` instead of 3?